tests/Service/FormatSponsorshipLevelBenefitTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: btaralle
+ * Date: 2019-02-07
+ * Time: 17:41
+ */
+
+namespace App\Tests\Service;
+
+use App\Repository\SponsorshipBenefit\SponsorshipBenefitManagerInterface;
+use App\Repository\SponsorshipLevel\SponsorshipLevelManagerInterface;
+use App\Repository\SponsorshipLevelBenefit\SponsorshipLevelBenefitManagerInterface;
+use App\Service\FormatSponsorshipLevelBenefit;
+use Prophecy\Argument;
+use PHPUnit\Framework\TestCase;
+
+class FormatSponsorshipLevelBenefitTest extends TestCase
+{
+    private $sponsorshipBenefitManagerProphecy;
+    private $sponsorshipLevelManagerProphecy;
+    private $sponsorshipLevelBenefitManagerProphecy;
+    private $formatSponsorshipLevelBenefit;
+
+    public function setUp()
+    {
+        $this->sponsorshipBenefitManagerProphecy = $this->prophesize(SponsorshipBenefitManagerInterface::class);
+        $this->sponsorshipLevelManagerProphecy = $this->prophesize(SponsorshipLevelManagerInterface::class);
+        $this->sponsorshipLevelBenefitManagerProphecy = $this->prophesize(SponsorshipLevelBenefitManagerInterface::class);
+
+        $this->formatSponsorshipLevelBenefit = new FormatSponsorshipLevelBenefit(
+            $this->sponsorshipBenefitManagerProphecy->reveal(),
+            $this->sponsorshipLevelManagerProphecy->reveal(),
+            $this->sponsorshipLevelBenefitManagerProphecy->reveal()
+        );
+    }
+
+    public function testFormat()
+    {
+        $this->sponsorshipLevelManagerProphecy->getOrderedList()->shouldBeCalled()->willReturn([]);
+        $this->sponsorshipBenefitManagerProphecy->getOrderedList()->shouldBeCalled()->willReturn([]);
+
+        $this->sponsorshipLevelBenefitManagerProphecy->getByBenefitAndLevel(Argument::any(), Argument::any())->shouldNotBeCalled();
```

Comment:
Should not be called? Do we have a misconception here? Should we introduce a more specific interface?

Reply:
for me is not misconception. I check the method is not call so i know i dont in the loops

Reply:
What I understand, if the function `getByBenefitAndLevel` should not be called, then it should not appear in the object itself, therefore the interface implemented in `sponsorshipBenefitManager` is not right and does not reflect the contrat, it might even need a more specific interface that extends other interfaces to allow a better decomposition of function to implement, or the `sponsorshipBenefitManager` can implement two interfaces